### PR TITLE
[dvswitch blocker] CLOUDSTACK-9591: Fix systemvmtemplate to not include network details

### PR DESCRIPTION
**Pull request by rhtyd**
This removes nic/network specific details while exporting the systemvmtemplate for vmware (ova file). Having this causes the ssvms to not deploy in dvswitch-based vmware environments that have no vswitch portgroups (dummy etc). Tested this on a local Trillian env.

### CODE REVIEW

**Comment by rhtyd** [comment]
@NuxRo hi Lucian, I tested your Macchinina template as well: http://dl.openvm.eu/cloudstack/macchinina/x86_64/macchinina-vmware.ova It too has network/nics defined so it fails to deploy on dvswitch-based vmware environments (with no vswitches). Can you please regenerate the ova without the defined nics/networks?

@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-55u3

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by NuxRo** [comment]
Hi,

I'll remove the NICs. Thanks for the notification.
I assume I'll have to do the same for all my templates as I am using the same script to generate the ova.

**Comment by rhtyd** [comment]
This may be seen as a blocker for vmware dv-switch based environments where at the moment we cannot use the systemvmtemplates.
Pinging for review - @DaanHoogland @abhinandanprateek @borisstoyanov @PaulAngus @karuturi @koushik-das and others

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-55u3

**Comment by karuturi** [comment]
I requested @sateesh-chodapuneedi for a review

File: vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java [hunk 1]
```diff
@@ -1477,6 +1493,40 @@ public static String resolveHostNameInUrl(DatacenterMO dcMo, String url) {
...
+            }
+            final DOMSource domSource = new DOMSource(doc);
+            final StringWriter writer = new StringWriter();
+            final StreamResult result = new StreamResult(writer);
+            final TransformerFactory tf = TransformerFactory.newInstance();
+            final Transformer transformer = tf.newTransformer();
+            transformer.transform(domSource, result);
+            return writer.toString();
+        } catch (SAXException | IOException | ParserConfigurationException | TransformerException e) {
+            s_logger.warn("Unexpected exception caught while removing network elements from OVF:", e);
```
Comment (DaanHoogland) [inline vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java]: thanks for this.
I'd log a warn without, and a debug with the stacktrace.That way an admin had better control on what is seen in the logs.

**Comment by karuturi** [comment]
@rhtyd can you check on the failures if they are related?

**Comment by rhtyd** [comment]
Thanks @karuturi the failures are not related to the PR.

**Comment by rhtyd** [comment]
@karuturi this is ready for merge

**Comment by sateesh-chodapuneedi** [comment]
>>Having this causes the ssvms to not deploy in dvswitch-based vmware environments that have no vswitch portgroups (dummy etc). 
@rhtyd can you please share the error/log messages for the deployment failures?

**Comment by rhtyd** [comment]
@sateesh-chodapuneedi see the JIRA ticket https://issues.apache.org/jira/browse/CLOUDSTACK-9591
The exceptions are seen in SSVM logs

**Comment by rhtyd** [comment]
Thanks @borisstoyanov 
@karuturi this is ready for merge now, thanks

**Comment by karuturi** [comment]
@rhtyd can you check on the failures? I think there are more failures on this PR. I usually see three failures.

**Comment by rhtyd** [comment]
@karuturi the failures around vpc are known to be intermittent for vmware env, the failures are not related to the changes. The environments where you dont see more than three failures are usually KVM based. I can kick the tests once again. 

@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
I don't think BO tests failures are related to this change. I think this is ready to merge.
cc: @karuturi

**Comment by karuturi** [comment]
ok. merging

**Pull request merged**